cirq-core/cirq/ops/clifford_gate.py

```diff
@@ -132,374 +131,260 @@ def _validate_map_input(
     return {frm: PauliTransform(to, flip) for frm, (to, flip) in pauli_map_to.items()}
 
 
-@value.value_equality
-class SingleQubitCliffordGate(gate_features.SingleQubitGate):
-    """Any single qubit Clifford rotation."""
+def _pad_tableau(
+    clifford_tableau: qis.CliffordTableau, num_qubits_after_padding: int, axes: List[int]
+) -> qis.CliffordTableau:
+    """Roughly, this function copies self.tabluea into the "identity" matrix."""
+    # Sanity check
+    if len(set(axes)) != clifford_tableau.n:
+        raise ValueError(
+            "Input axes of padding should match with the number of qubits in the input tableau."
+        )
+    if clifford_tableau.n > num_qubits_after_padding:
+        raise ValueError(
+            "The number of qubits in the input tableau should not be larger than "
+            "num_qubits_after_padding."
+        )
 
-    I = _pretend_initialized()
-    H = _pretend_initialized()
-    X = _pretend_initialized()
-    Y = _pretend_initialized()
-    Z = _pretend_initialized()
-    X_sqrt = _pretend_initialized()
-    Y_sqrt = _pretend_initialized()
-    Z_sqrt = _pretend_initialized()
-    X_nsqrt = _pretend_initialized()
-    Y_nsqrt = _pretend_initialized()
-    Z_nsqrt = _pretend_initialized()
+    padded_tableau = qis.CliffordTableau(num_qubits_after_padding)
+    v_index = np.concatenate((np.asarray(axes), num_qubits_after_padding + np.asarray(axes)))
+
+    padded_tableau.xs[np.ix_(v_index, axes)] = clifford_tableau.xs
+    padded_tableau.zs[np.ix_(v_index, axes)] = clifford_tableau.zs
+    padded_tableau.rs[v_index] = clifford_tableau.rs
+    return padded_tableau
 
-    def __init__(
-        self,
-        *,
-        _clifford_tableau: qis.CliffordTableau,
-    ) -> None:
-        self._clifford_tableau = _clifford_tableau
+
+class CommonCliffordGateMetaClass(value.ABCMetaImplementAnyOneOf):
+    """A metaclass used to lazy initialize several common Clifford Gate as class attributes."""
 
     @property
-    def clifford_tableau(self):
-        return self._clifford_tableau
+    def I(cls):
+        if getattr(cls, '_I', None) is None:
+            cls._I = cls._generate_clifford_from_known_gate(1, identity.I)
+        return cls._I
 
-    @staticmethod
-    def from_clifford_tableau(tableau: qis.CliffordTableau) -> 'SingleQubitCliffordGate':
-        assert isinstance(tableau, qis.CliffordTableau)
-        if not tableau._validate():
-            raise ValueError('It is not a valid Clifford Gate.')
-        return SingleQubitCliffordGate(_clifford_tableau=tableau)
+    @property
+    def X(cls):
+        if getattr(cls, '_X', None) is None:
+            cls._Z = cls._generate_clifford_from_known_gate(1, pauli_gates.X)
```

Comment:
Are these copy paste errors? The `_X`s, `_Y`s, and `_Z`s are all intermixed.

Reply:
Yes, great catch! Thanks!